stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecodePublicKey.java

```diff
@@ -0,0 +1,102 @@
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.stdlib.crypto.Constants;
+import org.ballerinalang.stdlib.crypto.CryptoUtils;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.cert.CertificateException;
+
+/**
+ * Function for decoding public key.
+ *
+ * @since 0.991.0
+ */
+@BallerinaFunction(
+        orgName = "ballerina", packageName = "crypto",
+        functionName = "decodePublicKey",
+        args = {
+                @Argument(name = "keyStore", type = TypeKind.RECORD, structType = Constants.KEY_STORE_RECORD),
+                @Argument(name = "keyAlias", type = TypeKind.STRING)
+        },
+        returnType = {
+                @ReturnType(type = TypeKind.RECORD, structType = Constants.PRIVATE_KEY_RECORD,
+                        structPackage = Constants.CRYPTO_PACKAGE),
+                @ReturnType(type = TypeKind.RECORD, structType = Constants.CRYPTO_ERROR,
+                        structPackage = Constants.CRYPTO_PACKAGE)
+        },
+        isPublic = true)
+public class DecodePublicKey extends BlockingNativeCallableUnit {
+
+    @Override
+    public void execute(Context context) {
+        BMap<String, BValue> keyStore = (BMap<String, BValue>) context.getNullableRefArgument(0);
+        BString keyAlias = (BString) context.getNullableRefArgument(1);
+
+        PublicKey publicKey = null;
+        // TODO: Add support for reading key from a provided string or directly using PEM encoded file.
+        if (keyStore != null) {
+            File keyStoreFile = new File(keyStore.get(Constants.KEY_STORE_RECORD_PATH_FIELD).stringValue());
+            try (FileInputStream fileInputStream = new FileInputStream(keyStoreFile)) {
+                KeyStore keystore = KeyStore.getInstance(Constants.KEYSTORE_TYPE_PKCS12);
+                try {
+                    keystore.load(fileInputStream, keyStore.get(Constants.KEY_STORE_RECORD_PASSWORD_FIELD).stringValue()
+                            .toCharArray());
+                } catch (NoSuchAlgorithmException e) {
+                    context.setReturnValues(CryptoUtils.createCryptoError(context,
+                            "keystore integrity check algorithm is not found: " + e.getMessage()));
+                    return;
+                }
+                publicKey = keystore.getCertificate(keyAlias.stringValue()).getPublicKey();
+
+                BMap<String, BValue> privateKeyStruct = BLangConnectorSPIUtil.createBStruct(context,
```

Comment:
```suggestion
                BMap<String, BValue> publicKeyStruct = BLangConnectorSPIUtil.createBStruct(context,
```
Shouldn't this be publicKeyStruct?